### PR TITLE
Update CFDataset to understand CMIP6 metadata format

### PR DESCRIPTION
**Pull request by corviday**
CMIP6 has renamed several of the required attributes since CMIP5. This PR adds the new attribute names that are updates of existing attribute names to `CFDataset`, along with a CMIP6 test case. In addition to pytest, these changes have been tested by successfully indexing a CMIP6-format file. 

A list of the renamed attributes can be seen [here](https://docs.google.com/document/d/1h0r8RZr_f3-8egBMMh7aqLwy3snpD6_MrDz1q8n5XUk), but most of them aren't used by nchelpers and so didn't need to be updated. I plan to also write a Confluence document listing all attributes we expect in a CMIP6 dataset- the official CMIP6 attributes, plus the additional PCIC ones we used with CMIP5 files.

### CODE REVIEW

**Review by rod-glover — CHANGES_REQUESTED** [review]
I think one (or both!) of us may have a misunderstanding of how the aliases work. See comments on the alias dicts. Let's discuss.

We also need some tests of the aliases against one or more CMIP6 files in `tests/test_CFDataset.py::test_metadata_simple_property`. And, depending on what decision we make about putting in new aliases for CMIP5, some new ones there too.

File: nchelpers/__init__.py [hunk 1]
```diff
@@ -579,7 +605,7 @@ def model_type(self):
         Really rudimentary decision making about model type.
         """
         if self.metadata.project == 'NARCCAP' or \
-                        self.metadata.project not in ('IPCC Fourth Assessment', 'CMIP5'):
+                        self.metadata.project not in ('IPCC Fourth Assessment', 'CMIP5', 'CMIP6'):
```
[the remark this answers is not in the document]
Comment (corviday) [reply nchelpers/__init__.py:608]: This whole file could use a blackify, but I was planning to do it as a separate PR.

File: tests/test_CFDataset.py [hunk 2]
```diff
@@ -367,6 +367,32 @@ def test_filepath(cwd, raw_dataset, converter, expected):
...
+    ('cmip6', 'is_streamflow_model_iobs_output', False),
+    ('cmip6', 'is_climdex_output', False),
+    ('cmip6', 'is_climdex_ds_gcm_output', False),
+    ('cmip6', 'is_gridded_obs', False),
+    ('cmip6', 'model_type', 'GCM'),
+    ('cmip6', 'ensemble_member', 'r1i1p1f1'),
+    ('cmip6', 'cmor_filename',
+     'pr_day_BCCAQv2_MPI-ESM1-2-HR_historical+ssp126_r1i1p1f1_19500101-19500131_Canada.nc'),
+    ('cmip6', 'unique_id',
+     'pr_day_BCCAQv2_MPI-ESM1-2-HR_historical-ssp126_r1i1p1f1_19500101-19500131_Canada'),
```
[the remark this answers is not in the document]
Comment (corviday) [reply tests/test_CFDataset.py:395]: I'll rename it `tiny_cmip6_gcm.nc`.

File: nchelpers/__init__.py [hunk 3]
```diff
@@ -426,6 +426,23 @@ def __init__(self, dataset):
...
+                'initialization_method': 'gcm.initialization_method',
+            },
+
+            'CMIP6': {
+                'project': 'project_id',
+                'initialization_method': 'gcm.initialization_index',
+                'physics_version': 'gcm.physics_index',
+                'realization': 'gcm.realization_index',
+                'forcing_index': 'gcm.forcing_index',
+                'realm': 'modeling_realm',
```
[the remark this answers is not in the document]
Comment (corviday) [reply nchelpers/__init__.py:440]: I think we actually don't use realm in nchelpers, I'll remove it.

File: nchelpers/__init__.py [hunk 4]
```diff
@@ -426,6 +426,23 @@ def __init__(self, dataset):
                 'institute': 'institute_id',
                 'experiment': 'gcm.experiment_id',
                 'ensemble_member': 'ensemble_member',  # uses prefixed values
+                'physics_version': 'gcm.physics_version',
+                'realization': 'gcm.realization',
+                'initialization_method': 'gcm.initialization_method',
+            },
+
+            'CMIP6': {
+                'project': 'project_id',
```
[the remark this answers is not in the document]
Comment (corviday) [reply nchelpers/__init__.py:435]: My thinking on this was twofold.

1) First, based on the sample values they gave for `activity_id`, I wasn't sure they were synonymous with our usage of `project_id`. They suggest "CMIP" as apossible value for `activity_id`but not a specific CMIP version as possible values. However, nchelpers needs a way to know what specific CMIP version a dataset adheres to.

2) Currently, nchelpers is hardwired to look for an attribute called `project_id` - as the value of that attribute that is used to determine which of these dictionaries to use.

[the remark this answers is not in the document]
Comment (corviday) [reply nchelpers/__init__.py:435]: After discussion, we have decided to switch from a single attribute to determine which (if any) version of CMIP a file is to a heuristic function

File: nchelpers/__init__.py [hunk 5]
```diff
@@ -504,6 +494,27 @@ def getdottedattr(obj, dotted_attr):
             except:
                 raise missing_attribute(attr)
 
+        def __get_metadata_standard__(self):
```
Comment (rod-glover) [inline nchelpers/__init__.py:497]: Eh, tiny comment here: I think "dunder" identifiers are (mainly) for magic methods like `__str__`, `__eq__`, etc. that the language uses to implement built-in functions and operators such as `str()`, `==`, etc. So maybe this should just be  `_get_metadata_standard`?

**Review by rod-glover — APPROVED** [review]
LGTM

**Pull request merged**